version: '3'

services:

  redis:
    image: redis
    ports:
      - '6379:6379'

  db:
    image: postgres
    environment:
      - POSTGRES_PASSWORD=mysecretpassword
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5


  app:
    build: .
    tty: true
    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py csu && python manage.py runserver 0.0.0.0:8000 "
    ports:
      - '8000:8000'
    volumes:
      - .:/app_drf
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    tty: true
    command: celery -A config worker -l info -P eventlet
    volumes:
      - .:/app_drf
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    restart: always
    tty: true
    command: celery -A config beat -l info -S django
    volumes:
      - .:/app_drf
    depends_on:
      - redis
      - app

volumes:
  pg_data:
